src/main.cpp

```diff
@@ -2455,8 +2455,11 @@ bool FindBlockPos(CValidationState &state, CDiskBlockPos &pos, unsigned int nAdd
     }
 
     nLastBlockFile = nFile;
-    vinfoBlockFile[nFile].nSize += nAddSize;
     vinfoBlockFile[nFile].AddBlock(nHeight, nTime);
+    if (fKnown && vinfoBlockFile[nFile].nSize < (pos.nPos + nAddSize))
+        vinfoBlockFile[nFile].nSize = pos.nPos + nAddSize;
```

Comment:
Looks good to me, although I'm not sure about the `fKnown && vinfoBlockFile[nFile].nSize >= (pos.nPos + nAddSize)` case. Would we want to increase nSize at all? Or is this better:
```c++
if (fKnown)
    vinfoBlockFile[nFile].nSize = std::max(pos.nPos + nAddSize, vinfoBlockFile[nFile].nSize);
else
    vinfoBlockFile[nFile].nSize += nAddSize;
```